flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -149,6 +154,9 @@ class JobManager(
 
   var leaderSessionID: Option[UUID] = None
 
+  private var metricsRegistry : FlinkMetricRegistry = _
```

Comment:
In theory this shouldn't be necessary as all metric-related code should never be able to crash anything.